cluster-autoscaler/core/scaledown/actuation/actuator_test.go

```diff
@@ -243,8 +244,10 @@ func TestCropNodesToBudgets(t *testing.T) {
 }
 
 func TestStartDeletion(t *testing.T) {
-	testNg := testprovider.NewTestNodeGroup("test-ng", 0, 100, 3, true, false, "n1-standard-2", nil, nil)
 	toBeDeletedTaint := apiv1.Taint{Key: deletetaint.ToBeDeletedTaint, Effect: apiv1.TaintEffectNoSchedule}
+	testNg := testprovider.NewTestNodeGroup("test-ng", 0, 100, 3, true, false, "n1-standard-2", nil, nil)
+	emptyNodeNodeGroup := generateNodesAndNodeGroupMap(4, "empty")
```

Comment:
We have one error message for all nodes from node group when there is failed deletion scenario for scale down with batcher. Also, if one deletion failed in one node group it may result of failed deletion for another node group. 
However, here we have batch with 0 second batching interval, so the nodes should be deleted one by one. But this is not always true, since we lock batcher in addNodeToBucket() and remove(), so actually we may add few nodes before remove() call. 
And this is what happening time to time, the current implementation make the test flaky. I fixed the test, to not be flaky, but now I'm not sure if it's the right approach. I would like to have an old behaviour (i.e. no batching) when the batch interval is 0 seconds.